src/utils/FixedPointMathLib.sol

```diff
@@ -120,56 +120,88 @@ library FixedPointMathLib {
     //////////////////////////////////////////////////////////////*/
 
     function sqrt(uint256 x) internal pure returns (uint256 result) {
-        if (x == 0) return 0;
+        assembly {
+            switch x
```

Comment:
i wonder if a not zero check would be cheaper than an switch?

Reply:
true true, one sec

Reply:
wow wtf that saved a lot more than i was expecting, i wanna take a look in the hevm debugger in the morning